common/transport/amqp/src/amqp.ts

```diff
@@ -48,70 +123,85 @@ export class Amqp {
     // In our case we want to control the reconnection flow ourselves, so we need to disable it.
 
-    /*Codes_SRS_NODE_COMMON_AMQP_16_042: [The Amqp constructor shall create a new `amqp10.Client` instance and configure it to:
+    /*Codes_SRS_NODE_COMMON_AMQP_16_042: [The Amqp constructor shall create a new `rhea.Client` instance and configure it to:
     - not reconnect on failure
-    - not reattach sender and receiver links on failure
-    - not reestablish sessions on failure]*/
-    this._amqp = new amqp10.Client(amqp10.Policy.merge(<any>{
-      session: {
-        reestablish: {
-          retries: 0,
-          forever: false
-        }
-      },
-      senderLink: {
-        attach: {
-          maxMessageSize: 0,
-        },
-        encoder: (body: any): any => {
-          if (typeof body === 'string') {
-            return new Buffer(body, 'utf8');
-          } else {
-            return body;
-          }
-        },
-        reattach: {
-          retries: 0,
-          forever: false
-        }
-      },
-      receiverLink: {
-        attach: {
-          maxMessageSize: 0,
-          receiverSettleMode: autoSettleMode,
-        },
-        decoder: (body: any): any => body,
-        reattach: {
-          retries: 0,
-          forever: false
-        }
-      },
-      // reconnections will be handled at the client level, not the transport level.
-      reconnect: {
-        retries: 0,
-        strategy: 'fibonacci',
-        forever: false
-      }
-    }, amqp10.Policy.EventHub));
+    - not reattach sender and receiver links on failure]*/
 
-    const amqpErrorHandler = (err) => {
-      debug('amqp10 client error: ' + err.toString());
+    this._rheaContainer = rheaCreateContainer();
+    this._rheaContainer.on('error', (context: EventContext) => {
+      debug('Container received a error event' + this._getErrorName(context.connection.error));
+    });
+
+    this._rheaContainer.on('azure-iot-amqp-base:error-indicated', (err: AmqpError) => {
+      debug('azure-iot-amqp-base:error-indicated invoked ' + this._getErrorName(err));
       this._fsm.handle('amqpError', err);
-    };
 
-    this._amqp.on('disconnected', () => {
-      // deferring this is necessary because in some instances
-      // the amqp10 library might want to send a close frame - and we don't want
-      // to trigger anything (especially not reconnection) before it has done so.
-      process.nextTick(() => {
-        this._fsm.handle('amqpDisconnected');
-      });
     });
 
+    // //
+    // // This error handler listens for events from THIS transport SenderLinks or ReceiverLinks.
+    // // These error indications would be as a result of an error indication from the associated
+    // // rhea links that are not associated with a specific request.
+    // //
+    // const amqpErrorHandler = (err) => {
+    //   debug('amqpErrorHandler invoked ' + this._getErrorName(err));
+    //   this._fsm.handle('amqpError', err);
+    // };
```

Comment:
remove?

Reply:
Done.

---
In reply to: [209130704](https://github.com/Azure/azure-iot-sdk-node/pull/344#discussion_r209130704) [](ancestors = 209130704)